internal/cli/atlas/quickstart/quick_start.go

```diff
@@ -284,6 +285,66 @@ func (opts *Opts) askClusterOptions() error {
 	return nil
 }
 
+func (opts *Opts) defaultRegions() []string {
+	cloudProviders, err := opts.store.CloudProviderRegions(opts.ConfigProjectID(), opts.Provider, tier, false)
+
+	if err != nil {
+		return DefaultRegions[strings.ToUpper(opts.Provider)]
+	}
+
+	if err := validateDefaultRegions(cloudProviders); err != nil {
+		return DefaultRegions[strings.ToUpper(opts.Provider)]
+	}
+
+	availableRegions := cloudProviders.Results[0].InstanceSizes[0].AvailableRegions
+
+	defaultRegions := make([]string, len(availableRegions))
+
+	// the most popular region must be the first in the list
+	popularRegion := ""
+	for _, v := range availableRegions {
+		if v.Default {
+			popularRegion = v.Name
+			break
+		}
+	}
+
+	defaultRegions[0] = popularRegion
+	i := 1
+
+	for _, v := range availableRegions {
+		if v.Name != popularRegion {
+			defaultRegions[i] = v.Name
+			i++
+		}
+	}
+
+	return defaultRegions
+}
+
+func validateDefaultRegions(cloudProviders *atlas.CloudProviders) error {
```

Comment:
[q] this seems like a test method? why would users need to see this?

Reply:
What do you mean with  
> why would users need to see this?
The user doesn’t see anything, I use this function to ensure that I can access the array with indexes later

Reply:
oh I see you don't do anything with the text of the error, so why so much effort on the error message?
Anyway my original questions stand, why test the API result? are you not able to trust the API?

Reply:
I don't like using this piece of code `cloudProviders.Results[0].InstanceSizes` without testing for possible NPE. No, I don't trust my code 😅 .

Reply:
you'll get nil panics or NPE if the API returns an error. again this code reads like a test, so maybe this is probably not the best place for this code

use `len(cloudProviders.Results) > 0 && len(cloudProviders.Results[0].InstanceSizes) > 0` or soem combination of it `len(nil) ==0` btw